### ISSUE TITLE
UI Localization issue

### ISSUE THREAD

**Issue by huguesbr** [ParsePlatform/ParseUI-iOS#192]
Localization work only for some element.

I've added a `ParseUI.localizable` in french to my project.
I've copied/paste the full content of the english version distributed with the pod.
I've translated all strings (happy to add it to base).

    "Reset Password" = "Reset Password";

"Sign Up" = "Sign Up";

**Comment by huguesbr** [comment]
BTW, isn't it a bit hardcore to rename the `NSLocalizedString` macro?
Wouldn't be better to just create a new macro `PFLocalizedString`?

**Comment by nlutsenko** [comment]
`PFLocalizedString` - yeah, probably a better solution. Will submit a PR in a bit.
Not sure on why `Sign Up` button test is not translated, just searched through the code base and it looks like every usage of that is localized.

**Comment by huguesbr** [comment]
@nlutsenko I can do the update for `PFLocalizedString`

**Comment by nlutsenko** [comment]
@huguesbr, awesome! Be my guest, would gladly review and accept the Pull Request.

**Comment by huguesbr** [comment]
@nlutsenko I think it have to do with using the pod as a framework, which then encapsulate the file. If I tweak the remap macro and use `[NSBundle mainBundle]` then it work like a charm.

**Comment by nlutsenko** [comment]
Huh, this is strange... Since the macro expands into `bundleForClass:` it should just find the appropriate resources...

@huguesbr, do you add the localization to the main bundle or the framework bundle itself?
(you need to do it in the framework bundle)

**Comment by huguesbr** [comment]
@nlutsenko using cocoa pods how can I do this?

btw, PR ready https://github.com/ParsePlatform/ParseUI-iOS/pull/193

**Comment by huguesbr** [comment]
@nlutsenko my point is that I know how to add this file to the bundle `ParseUI` by dropping it in the `Pods` project and selecting the right Pod. But then it will be trash next time I'm doing `pod install` (my guess). I guess it might be more a `cocoapod` question, but would love if you have any idea...

**Comment by nlutsenko** [comment]
@huguesbr, yup, got you.
Got curious myself and did a little research.
Turns out - it's not as easy as it seems!

The easiest and probably the most stable options would be:
- Fork the project (even private one), add the localization files to the fork and update your Podfile to use the version of ParseUI from your fork
- Create a small script that will do the `pod update`/`pod install` as well as copy/paste the resources into the proper folder for you

The whole idea here comes down to the fact, that there is no easy way to modify a podspec at the process of integration.

**Comment by huguesbr** [comment]
@nlutsenko yep, I've done the step 1.. I will submit a PR, so maybe you can directly integrate french :)
I will keep digging into cocoa pods, I'm sure there is a way to inject a file or something...

**Comment by nlutsenko** [comment]
There is definitely a hacky way to do this, I wonder whether there is one that is actually stable...

**Comment by huguesbr** [comment]
yeah, well meanwhile, here is the PR to translate it in french.. https://github.com/ParsePlatform/ParseUI-iOS/pull/195

**Comment by huguesbr** [comment]
@nlutsenko not sure what to do about this issue. it's still problematic for anybody who would like translate the UI.

**Comment by sibeliusseraphini** [comment]
Maybe this can help: http://stackoverflow.com/questions/12874678/localization-and-cocoapods

**Comment by huguesbr** [comment]
@sibeliusseraphini the first 2 solutions are pretty insanely ugly and hacky.. check for existence each time you use a localized string seem pretty heavy.. last solution to inject is pretty nice but damn, wish there was a simpler way. I will give it a shot.

**Comment by estemendoza** [comment]
Thanks for you hard work. Quick questions, if I don't install this using cocoa pods, I just add the source files to my project and compile, how can I make the translation to another language?

BTW, I am forking to commit the spanish version

**Comment by huguesbr** [comment]
@estemendoza You should be able to just create a new adding new translation for `ParseUI.strings`

**Comment by lastMove** [comment]
I didn't understand : How should we do to change localisation when using the "ParseUI.framework" directly (without cocoa pods) ?

**Comment by huguesbr** [comment]
@lastMove can you elaborate? Are you using `Carthage`? if you want to add a localisation, the best way would be to fork this repo, add your translated `ParseUI.strings` to the repo (check this PR https://github.com/ParsePlatform/ParseUI-iOS/pull/195) and then you can use your fork until your PR is accepted and merge to the main repo.

**Comment by lastMove** [comment]
Hi, Thanks for your answer. 
Yes I am using carthage in my project, but I installed ParseUI by adding the "ParseUI.framework" (found [here](http://parse-ios.s3.amazonaws.com/00e620d6f8c3fd9f167fa0d297ed723d/parse-library-1.12.0.zip) ). 
I need the french localisation which is already included (by you, thanks :) ). 
But since i am not using cocoa pods, it's not working. All texts are in english.

**Comment by huguesbr** [comment]
What if you manually add the french `ParseUI.strings` to your project?

**Comment by lastMove** [comment]
Ah, I will try it thank you
